src/controller/CHIPDeviceController.cpp

```diff
@@ -228,12 +234,68 @@ CHIP_ERROR DeviceController::Init(NodeId localDeviceId, ControllerInitParams par
     mState         = State::Initialized;
     mLocalDeviceId = localDeviceId;
 
+    VerifyOrExit(params.operationalCredentialsDelegate != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
+    mOperationalCredentialsDelegate = params.operationalCredentialsDelegate;
+
+    err = LoadLocalCredentials(admin);
+    SuccessOrExit(err);
+
     ReleaseAllDevices();
 
 exit:
     return err;
 }
 
+CHIP_ERROR DeviceController::LoadLocalCredentials(Transport::AdminPairingInfo * admin)
+{
+    ChipLogProgress(Controller, "Getting operational keys");
+    Crypto::P256Keypair * keypair = admin->GetOperationalKey();
+
+    ReturnErrorCodeIf(keypair == nullptr, CHIP_ERROR_NO_MEMORY);
+
+    if (!admin->AreCredentialsAvailable())
+    {
+        chip::Platform::ScopedMemoryBuffer<uint8_t> buffer1;
+        ReturnErrorCodeIf(!buffer1.Alloc(kMaxCHIPOpCertLength), CHIP_ERROR_NO_MEMORY);
+
+        chip::Platform::ScopedMemoryBuffer<uint8_t> buffer2;
+        ReturnErrorCodeIf(!buffer2.Alloc(kMaxCHIPOpCertLength), CHIP_ERROR_NO_MEMORY);
+
+        uint8_t * CSR    = buffer1.Get();
+        size_t csrLength = kMaxCHIPOpCertLength;
```

Comment:
The CSR length has nothing to do with an opcert length.

A CSR contains mostly just a public key and signature and some flags (e.g. extensions), in PCKS10 DER format, which is quite a bit less than an actual opcert.

I think it's OK for the short term, but a `kMaxCHIPCsrLength` would be worthwhile to define, even if it matches `kMaxCHIPOpCertLength` for now.